src/backend/DocDownloaderRedisCache/src/main/java/ca/bc/gov/ag/courts/controller/JobController.java

```diff
@@ -0,0 +1,84 @@
+package ca.bc.gov.ag.courts.controller;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import ca.bc.gov.ag.courts.model.Job;
+import ca.bc.gov.ag.courts.repo.JobRepository;
+
+@RestController
+public class JobController {
+
+	Logger logger = LoggerFactory.getLogger(JobController.class);
+
+	@Autowired
```

Comment:
Personally I prefer using the constructors, however since this code was already done by @shaunmillargov using Autowired so kept it as is to reduce effort. However I can try to rewrite if you feel strongly.

Reply:
Is there an appreciative advantage of this @eugenezimin? The application works. I have no issue with a switching to your suggestion if 1.) It doesn't take long for re-coding, and 2.) The service remains as a singleton.

Reply:
Not too much Shaun, I would say both of them are workable solutions, it is rather closer to the concept of clean code. Don't think I insist on this and block from going further - it's rather a food for thinking.

Benefits of using DI through constructor are the following:

- it actually promotes the IoC principle itself, where the class explicitly declares its dependencies in the constructor
- it also ensures that the required dependencies are available when the object is created, preventing NPE and other errors (fail-fast)
- it makes the code more explicit and easier to understand, as dependencies are clearly visible in the constructor signature
- it facilitates unit testing by allowing dependencies to be easily mocked or replaced
- and it prevents injection of partially initialized objects - which I would say a great feature to have for debugging as it is possible to trace the full call stack easily, rather then figure out where `Autowired` is initialized.

Again - it's rather thoughts aloud than even a call for action.